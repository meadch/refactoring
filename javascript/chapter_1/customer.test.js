const Customer = require('./customer')
const { Movie } = require('./movie')
const Rental = require('./rental')

const secondMovie = new Movie('freddy', 1)
const firstMovie = new Movie('silence of the lambs', 0)
const kidsMovie = new Movie('Finding Nemo', 2)
const kidsMovie2 = new Movie('Finding Nemo 2', 2)

const firstRental = new Rental(firstMovie, 2)
const secondRental = new Rental(secondMovie, 4)
const kidsMovieRental = new Rental(kidsMovie, 3)
const kidsMovieRental2 = new Rental(kidsMovie2, 7)

test('can create a create a customer', () => {
  const customer = new Customer('bob')
  expect(customer.getName()).toBe('bob')
})

test('customers can rent movies and get a proper receipt', () => {
  const customer = new Customer('bob')
  customer.addRental(firstRental)
  customer.addRental(secondRental)
  customer.addRental(kidsMovieRental)
  customer.addRental(kidsMovieRental2)
  expect(customer.statement()).toMatchSnapshot()
})

test('customers can rent movies and get a proper receipt in html format', () => {
  const customer = new Customer('bob')
  customer.addRental(firstRental)
  customer.addRental(secondRental)
  customer.addRental(kidsMovieRental)
  customer.addRental(kidsMovieRental2)
  expect(customer.htmlStatement()).toMatchSnapshot()
})
