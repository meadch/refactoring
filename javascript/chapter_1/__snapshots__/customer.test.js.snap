// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customers can rent movies and get a proper receipt 1`] = `
"Rental Record for bob 
	 silence of the lambs 	 2 
	 freddy 	 12 
	 Finding Nemo 	 1.5 
	 Finding Nemo 2 	 7.5 
Amount owed is: 23
    You earned 5 frequent renter points."
`;

exports[`customers can rent movies and get a proper receipt in html format 1`] = `
"<h1>Rentals for <em> bob </em></h1>
<p>silence of the lambs: 2 </p>
<p>freddy: 12 </p>
<p>Finding Nemo: 1.5 </p>
<p>Finding Nemo 2: 7.5 </p>
<p>You owe <em>23</em></p>
<p><You earned <em>5</em> frequent renter points</p>"
`;
